src/test/deploy/functions/pricing.spec.ts

```diff
@@ -0,0 +1,299 @@
+import { expect } from "chai";
+import { v1 } from "uuid";
+
+import * as backend from "../../../deploy/functions/backend";
+import * as pricing from "../../../deploy/functions/pricing";
+
+const FUNCTION_FRAGMENT: Omit<backend.FunctionSpec, "apiVersion" | "region"> = {
+  id: "function",
+  project: "project",
+  entryPoint: "foobar",
+  runtime: "nodejs14",
+  trigger: {
+    allowInsecure: false,
+  },
+};
+
+const INVALID_REGION = { region: "fillory" };
+describe("Functions Pricing", () => {
```

Comment:
These tests feel weird to me. On one hand, I like that we are thoroughly testing this. On the other, this feels a bit circular (We're testing our calculations by basically copying part of the calculations code into a test file), and I don't like the magic numbers peppered throughout (it took me a minute or two to realize that many of these were just numbers from MB_TO_GHZ).

I don't know if this is really a better option (since it would be obnoxious to maintain), but we could simplify this and make it less circular by hard coding how much each case should cost. WDYT?

Reply:
The tests actually caught bugs. The trick is to redo as much arithmetic as possible from scratch without copy + paste. Ideally you'd calculate in a slightly different method. If two independent implementations come up with the same result it's more likely correct. One of my advisors actually did this for oil drill safety checks; he had fully independent teams maintain separate calculation implementations and the app constantly checked for the two results to be in sync.